src/components/Editor/index.js

```diff
@@ -104,80 +116,111 @@ type State = {
 
 class Editor extends PureComponent<Props, State> {
   $editorWrapper: ?HTMLDivElement;
+  emptyLineDecorations: any[];
+
   constructor(props: Props) {
     super(props);
 
     this.state = {
       highlightedLineRange: null,
       editor: null
     };
+
+    this.emptyLineDecorations = [];
   }
 
   componentWillReceiveProps(nextProps) {
     if (!this.state.editor) {
       return;
     }
 
-    resizeBreakpointGutter(this.state.editor.codeMirror);
-    resizeToggleButton(this.state.editor.codeMirror);
+    resizeToggleButton(this.state.editor.monaco);
   }
 
   componentWillUpdate(nextProps) {
     this.setText(nextProps);
     this.setSize(nextProps);
+    this.setEmptyLines(nextProps);
     this.scrollToLocation(nextProps);
   }
 
   setupEditor() {
-    const editor = getEditor();
-
-    // disables the default search shortcuts
-    // $FlowIgnore
-    editor._initShortcuts = () => {};
+    const editor = new SourceEditor({
+      theme: "vs",
```

Comment:
@jasonLaster , any idea of how I can access the current theme option? It seems that themes are controlled by css but not options directly, or maybe I miss something.